clientconn_parsed_target_test.go

```diff
@@ -42,46 +42,46 @@ func (s) TestParsedTarget_Success_WithoutCustomDialer(t *testing.T) {
-		{target: "unix-abstract:///a/b/c", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", URL: *testutils.MustParseURL("unix-abstract:///a/b/c")}},
-		{target: "unix-abstract:///", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", URL: *testutils.MustParseURL("unix-abstract:///")}},
-		{target: "passthrough:///unix:///a/b/c", wantParsed: resolver.Target{Scheme: "passthrough", Authority: "", URL: *testutils.MustParseURL("passthrough:///unix:///a/b/c")}},
+		{target: "dns:///google.com", wantParsed: resolver.Target{URL: *testutils.MustParseURL("dns:///google.com")}},
+		{target: "dns://a.server.com/google.com", wantParsed: resolver.Target{URL: *testutils.MustParseURL("dns://a.server.com/google.com")}},
+		{target: "dns://a.server.com/google.com/?a=b", wantParsed: resolver.Target{URL: *testutils.MustParseURL("dns://a.server.com/google.com/?a=b")}},
+		{target: "unix:///a/b/c", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix:///a/b/c")}},
+		{target: "unix-abstract:a/b/c", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:a/b/c")}},
+		{target: "unix-abstract:a b", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:a b")}},
+		{target: "unix-abstract:a:b", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:a:b")}},
+		{target: "unix-abstract:a-b", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:a-b")}},
+		{target: "unix-abstract:/ a///://::!@#$%25^&*()b", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:/ a///://::!@#$%25^&*()b")}},
+		{target: "unix-abstract:passthrough:abc", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:passthrough:abc")}},
+		{target: "unix-abstract:unix:///abc", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:unix:///abc")}},
+		{target: "unix-abstract:///a/b/c", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:///a/b/c")}},
+		{target: "unix-abstract:///", wantParsed: resolver.Target{URL: *testutils.MustParseURL("unix-abstract:///")}},
+		{target: "passthrough:///unix:///a/b/c", wantParsed: resolver.Target{URL: *testutils.MustParseURL("passthrough:///unix:///a/b/c")}},
 
 		// Cases for `scheme:absolute-path`.
-		{target: "dns:/a/b/c", wantParsed: resolver.Target{Scheme: "dns", Authority: "", URL: *testutils.MustParseURL("dns:/a/b/c")}},
-		{target: "unregistered:/a/b/c", badScheme: true, wantParsed: resolver.Target{Scheme: defScheme, Authority: "", URL: *testutils.MustParseURL("unregistered:/a/b/c")}},
+		{target: "dns:/a/b/c", wantParsed: resolver.Target{URL: *testutils.MustParseURL("dns:/a/b/c")}},
+		{target: "unregistered:/a/b/c", badScheme: true, wantParsed: resolver.Target{URL: *testutils.MustParseURL("unregistered:/a/b/c")}},
```

Comment:
If you look below..... `wantParsed.URL` is overwriting this.

We should either delete that code or remove all of these manual calls to set it.  Either way is probably okay.

Also, I'm not sure this test has all that much value anymore.  We can probably remove 75% of these test cases since nothing is special-cased anymore and everything is parsed with `url.Parse`.